### DISCUSSION TITLE
You're genius!

### DISCUSSION THREAD

**Discussion by balintn22** [matt-goldman/Maui.Plugins.PageResolver#21] [General]
Thank you!
I suffered so much with Xamarin Forms / Maui navigation, and especially with argument passing. Navigation properties might work if you do everything correctly, but if you don't, all you experience is a failure. Maui shell navigation and the out of the box navigation-property based idea is nice, as it looks like URLs, but it is as fragile as Maui in general.

Your idea about piggybacking navigation on DI is genius! It works and the error message when I fail to pass an argument is helpful, makes troubleshooting a breeze.

(Japanese bow)

**Comment by matt-goldman** [comment]
Thank you so much for the kind words! I'm glad you find this library useful. I shared your frustrations with Xamarin Forms which is what ultimately lead me to share this library. I'm happy to see other people are enjoying it too :)
